Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-{
-	bool moveSuccessful = false;
-	int i = 0;
-
-	i = current_Room->directionIndex(direction);
-
-	if ((i == 3) && ((current_Room->getRoomNumber()) == 1))
-	{
-		exit_Castle = true;
-	}
-
-	else if((current_Room->getPaths())[i] != nullptr)
-	{
-		current_Room   = (current_Room->getPaths())[i];
-		moveSuccessful = true;
-
-		if(current_Room->getPrincessPresent())
-		{
-			HelperFunctions::color(AQUA);
-			cout << "The Princess is standing , waiting for you in the dark . ";
-			princess_Pointer= current_Room->getPrincessPresent();
-			current_Room->setPrincessPresent(nullptr);
-		}
-
-		HelperFunctions::color(RED);
-		cout << "\nYou have successfully been moved to Room : " << current_Room->getRoomNumber();
-	}
-
-	else
-	{
-		HelperFunctions::color(RED);
-		cout << "\nMOVE " << direction <<" is an invalid command.";
-	}
-
-	return moveSuccessful;
-}
-
-
-
-
-
-
-
-void Player::Pick(char const* itemName)
-{
-	bool picked = false;
-	bool itemFound = false;
-
-	for (int i = 0; i < 5; ++i)
-	{
-		if((current_Room->getItemsPresent())[i] != nullptr)
-		{
-			if(	strcmp((((current_Room->getItemsPresent())[i])->getItemName()) , itemName) == 0)
-			{
-				itemFound = true;
-
-				if (!isBagFull())
-				{
-					int j = 0;
-
-					for (; bag[j] != nullptr; ++j)
-					{
-
-					}
-
-					bag[j] =  (current_Room->getItemsPresent())[i];
-					current_Room->setItemsPresent(i , nullptr);
-
-					picked = true;
-
-					CashUpdater();
-				}
-				else
-					cout << "\nSorry , your bag is already full.";
-			}
-		}
-	}
-
-	HelperFunctions::color(RED);
-	if(picked)
-		cout <<endl << itemName << " has successfully been picked.";
-
-	else if(itemFound == false)
-		cout <<endl << itemName <<" is not present in this room.";
-}
-
-
-
-
-void Player::Drop(char const* itemName)
-{
-	bool dropped = false;
-	bool item_in_bag = false;
-
-	for (int i = 0; i < 10; ++i)
-	{
-		if(bag[i] != NULL)
-		{
-			if(	strcmp((bag[i])->getItemName() , itemName) == 0)
-			{
-				item_in_bag = true;
-
-				if(!(current_Room-> isRoomFull()))
-				{
-					int j = 0;
-
-					for (; (current_Room->getItemsPresent())[j] != nullptr; ++j)
-					{
-
-					}
-
-					current_Room->setItemsPresent(j , bag[i]);
-					bag[i] = nullptr;
-
-					dropped = true;
-
-					CashUpdater();
-				}
-				else
-				{
-					HelperFunctions::color(AQUA);
-					cout << "/n/The item" << itemName << " couldn't be dropped as Room "<<current_Room->getRoomNumber()
-						 << " is already full.";
-				}
-			}
-		}
-	}
-
-	HelperFunctions::color(YELLOW);
-
-	if(dropped)
-		cout << endl<< itemName <<" has successfully been dropped.";
-
-	else if(item_in_bag == false)
-		cout << endl << itemName <<" couldn't be dropped as it is not present in your bag.";
-}
-
-
-
-
-
-bool Player::Attack(char const* monsterName)
-{
-	bool killed = false;
-
-	if(current_Room->getMonsterPresent() != nullptr)
-	{
-		if(((current_Room->getMonsterPresent())->getLivingState()) == true)
-		{
-			for (int i = 0; i < 10; ++i)
```

Comment:
Why is i looping from 0 to 9? Is there any significance?  10 should not be hardcoded in the loop.